### ISSUE TITLE
Layers disappear on zoom

### ISSUE THREAD

**Issue by nstjean** [publiclab/leaflet-environmental-layers#399]
I noticed a bug: This map works great when `simpleMenu: true` but when the browser menu is used the layers are disappearing on zoom.

![Peek 2020-02-11 11-55](https://user-images.githubusercontent.com/49460529/74259256-9fb61500-4cc5-11ea-95a7-f589ce28d12a.gif)

Here's the code I used in `multipleMaps.html`:
```
      var bounds = new L.LatLngBounds(
        new L.LatLng(84.67351257, -172.96875),
        new L.LatLng(-54.36775852, 178.59375)
      );

      var map = L.map("map", {
        maxBounds: bounds,
        maxBoundsViscosity: 0.75,
      }).setView([43, -83], 3);
      map.options.minZoom = 3;

      var map2 = L.map("map2", {
        maxBounds: bounds,
        maxBoundsViscosity: 0.75,
      }).setView([43, -83], 3);
      map2.options.minZoom = 3;

      L.LayerGroup.EnvironmentalLayers({
        include: ['odorreport', 'asian', 'clouds', 'eonetFiresLayer', 'Unearthing', 'PLpeople'],
        addLayersToMap: true,
        // simpleLayerControl: true,
      }).addTo(map);

      L.LayerGroup.EnvironmentalLayers({
        // include: ['odorreport', 'asian', 'clouds', 'eonetFiresLayer', 'Unearthing', 'PLpeople'],
        // display: ['odorreport', 'clouds', 'eonetFiresLayer', 'Unearthing'], 
      }).addTo(map2);
```

I also noticed that those layers aren't listed at all in the browser menu. But this might just be a result of the changes we made in #396

**Comment by nstjean** [comment]
@crisner

**Comment by crisner** [comment]
That is because it is filtered out by the layer menu. It is not supposed to be visible at that map bounds or zoom. It appears because we are forcing it to display on the map by default.

**Comment by nstjean** [comment]
Ahhhh. Ok. I see it did behave this way before our changes .

**Comment by crisner** [comment]
When a user is browsing the layer will only be available in the layer's zoom and bounds. So a user might set it to appear by default only when the layer is available. It would disappear when the map moves away from its bounds or the map is zoomed out.

**Comment by nstjean** [comment]
I do see that with `simpleLayerControl: true` the map `asian` does disappear at zoom 10 and higher. Which is expected. But when `simpleLayerControl: false` it disappears as soon as I zoom the map at all - and it does not turn back on.

**Comment by crisner** [comment]
It gets removed from the map. Justice map is available from zoom level 5. You will have to turn it back on when it is available in the layer menu.

**Comment by crisner** [comment]
If this seems unintuitive we can remove the code responsible for removing the layer from the map. The issue then would be there would be no way to turn off the layer when it is filtered off the layer menu.

**Comment by crisner** [comment]
We should perhaps look into a better way for layer filtering? Maybe look into how the layer should behave with the map when it is filtered out?

**Comment by nstjean** [comment]
Ahhh, I see. Yeah that doesn't seem intuitive to me.

So it behaves differently depending on which menu is up because the browser menu doesn't have all the layers in there? I noticed a lot are missing, such as Unearthing.

**Comment by crisner** [comment]
Yes. The layer menu in LEL is different from the default one. We essentially want the users to work with LEL's layer menu. The layers are filtered out because we don't want the user requesting data out of the layer's extents and we don't want the layer menu to be too long.

Here's how we could go about making the experience better. 

1) Once a layer is added to the map let the layer be visible both on the map and the layer menu but keep the layer from requesting data if the map is not intersecting with the layer's extents.

2) The layers that are not displayed on the map will be filtered from the layer menu. We could add a feature that lets the user decide if they want to keep the layers filtered or if they want everything to be listed.

3) We have an alert that notifies users of new layers available. We may have to highlight the new layers as well so the users can locate the new layers easily.

@nstjean, @jywarren, and @sagarpreet-chadha, What do you think?

**Comment by nstjean** [comment]
Ohhhhhh, I see!! That is neat!

I don't see the Wisconsin layer showing up at all, even when I zoom in to Wisconsin, though. We will want to check into that and make sure the visibility settings are correct.

1. That makes sense.
2. I like that idea, a toggle button in the menu to "show all layers"
3. Yes I like that idea of highlighting the new ones! It's hard to tell right now.

**Comment by crisner** [comment]
> I don't see the Wisconsin layer showing up at all, even when I zoom in to Wisconsin, though. We will want to check into that and make sure the visibility settings are correct.

Yes. I've come across that too. The data is requested only when map view is on 'Eau Claire'. Not sure why it was coded that way. Probably because the data is not available in other parts of Wisconsin? We'll need to check if that's the case.

**Comment by nstjean** [comment]
It seems to only show if you are zoomed in really close. We can maybe adjust that.

**Comment by sagarpreet-chadha** [comment]
Hey,
The above layer and all similar layers (asian, etc.) are fetched only once, so let's not make them disappear abruptly.

**Comment by crisner** [comment]
Fixed in #409. We will no longer be removing layers that are displayed on the map. 😄
